File: src/postfixcalculator/ConsoleInputFetcher.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package postfixcalculator;

import java.util.Scanner;

/**
 *
 * @author d.peters
 */
public class ConsoleInputFetcher implements IInputFetcher{
    
    /**
     * 
     */
    private Scanner scanner;
    
    /**
     * 
     */
    public ConsoleInputFetcher(){
        this.scanner = new Scanner(System.in);
    }
    
    /**
     * 
     * @return 
     */
    @Override
    public String getInput() {
        return this.scanner.nextLine();
    }
    
}

File: src/postfixcalculator/ConsoleInputFetcher.kt
package postfixcalculator

import java.util.Scanner

/**
 *
 * @author Daniel Peters
 * @version 1.0
 */
class ConsoleInputFetcher : IInputFetcher {
  private val scanner = Scanner(System.`in`)

  override fun getInput(): String {
    return this.scanner.nextLine()
  }
}
